homeassistant/components/mobile_app/webhook.py

```diff
@@ -0,0 +1,192 @@
+
+    webhook_payload = req_data.get(ATTR_WEBHOOK_DATA, {})
+
+    if req_data[ATTR_WEBHOOK_ENCRYPTED]:
+        enc_data = req_data[ATTR_WEBHOOK_ENCRYPTED_DATA]
+        webhook_payload = _decrypt_payload(device[CONF_SECRET], enc_data)
+
+    if webhook_type not in WEBHOOK_TYPES:
+
+        if ATTR_APP_COMPONENT not in device:
+            _LOGGER.error("Unknown mobile_app webhook type: %s", webhook_type)
+            return empty_okay_response(headers=headers)
+
+        # Unknown webhook type, check if there's a component
+        platform_name = device[ATTR_APP_COMPONENT]
+
+        plat = get_platform(hass, platform_name, DOMAIN)
+
+        if webhook_type in plat.WEBHOOK_TYPES:
+            return await plat.async_handle_webhook_message(hass, device,
+                                                           webhook_type,
+                                                           webhook_payload)
+
+    try:
+        data = WEBHOOK_SCHEMAS[webhook_type](webhook_payload)
+    except vol.Invalid as ex:
+        err = vol.humanize.humanize_error(webhook_payload, ex)
+        _LOGGER.error('Received invalid webhook payload: %s', err)
+        return empty_okay_response(headers=headers)
+
+    if webhook_type == WEBHOOK_TYPE_CALL_SERVICE:
+        try:
+            await hass.services.async_call(data[ATTR_DOMAIN],
+                                           data[ATTR_SERVICE],
+                                           data[ATTR_SERVICE_DATA],
+                                           blocking=True,
+                                           context=device_context(device))
+        # noqa: E722 pylint: disable=broad-except
+        except (vol.Invalid, ServiceNotFound, Exception) as ex:
+            _LOGGER.error("Error when calling service during mobile_app "
+                          "webhook (device name: %s): %s",
+                          device[ATTR_DEVICE_NAME], ex)
+            raise HTTPBadRequest()
+
+        return empty_okay_response(headers=headers)
+
+    if webhook_type == WEBHOOK_TYPE_FIRE_EVENT:
+        event_type = data[ATTR_EVENT_TYPE]
+        hass.bus.async_fire(event_type, data[ATTR_EVENT_DATA],
+                            EventOrigin.remote,
+                            context=device_context(device))
+        return empty_okay_response(headers=headers)
+
+    if webhook_type == WEBHOOK_TYPE_RENDER_TEMPLATE:
+        try:
+            tpl = template.Template(data[ATTR_TEMPLATE], hass)
+            rendered = tpl.async_render(data.get(ATTR_TEMPLATE_VARIABLES))
+            return json_response({"rendered": rendered}, headers=headers)
+        # noqa: E722 pylint: disable=broad-except
+        except (ValueError, TemplateError, Exception) as ex:
+            _LOGGER.error("Error when rendering template during mobile_app "
+                          "webhook (device name: %s): %s",
+                          device[ATTR_DEVICE_NAME], ex)
+            return json_response(({"error": ex}), status=HTTP_BAD_REQUEST,
```

Comment:
`ex` is not JSON serializable.

Reply:
when do you expect `ValueError` or `Exception` ? You can drop those.

Reply:
`ValueError` was pulled from the `/api/template` logic directly. `Exception` got added so that any and all errors get caught and properly errored out, you asked to add it [here](https://github.com/home-assistant/home-assistant/pull/21607#discussion_r261928123).

Wrapped ex with `str`.